test/e2e/framework.go

```diff
@@ -161,6 +161,21 @@ func (f *Framework) afterEach() {
 		summaries = append(summaries, f.logsSizeVerifier.GetSummary())
 	}
 
+	if testContext.GatherMetricsAfterTest {
+		// TODO: enable Scheduler and ControllerManager metrics grabbing when Master's Kubelet will be registered.
+		grabber, err := metrics.NewMetricsGrabber(f.Client, true, false, false, true)
+		if err != nil {
+			Logf("Failed to create MetricsGrabber. Skipping metrics gathering.")
+		} else {
+			received, err := grabber.Grab(nil)
+			if err != nil {
+				Logf("MetricsGrabber failed grab metrics. Skipping metrics gathering.")
+			} else {
+				summaries = append(summaries, (*metricsForE2E)(&received))
```

Comment:
Yup. received has a type MetricsCollection which does not have PrintHumanReadable method, and metricsForE2E implements Summary interface (i.e. implement PrintHumanReadable and PrintJSON). It's a simple 'wrapper' type for MetricsCollection that adds those methods to them.